### PR TITLE
Misc fixes from runtimelab:NativeAOT

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/libraries/Native/Windows/System.IO.Compression.Native/CMakeLists.txt [hunk 1]
```diff
@@ -95,7 +95,6 @@ add_library(System.IO.Compression.Native-static
     ${NATIVECOMPRESSION_SOURCES}
 )
 
-SET_TARGET_PROPERTIES(System.IO.Compression.Native-static PROPERTIES PREFIX "")
 SET_TARGET_PROPERTIES(System.IO.Compression.Native-static PROPERTIES OUTPUT_NAME System.IO.Compression.Native)
```
Comment (jkotas) [inline src/libraries/Native/Windows/System.IO.Compression.Native/CMakeLists.txt:98]: cc @VSadov The default `PREFIX` on Windows is empty string. This was no-op.

File: src/libraries/Native/Windows/System.IO.Compression.Native/CMakeLists.txt [hunk 2]
```diff
@@ -123,4 +122,6 @@ if (GEN_SHARED_LIB)
 endif()
 
 install (TARGETS System.IO.Compression.Native-static DESTINATION ${STATIC_LIB_DESTINATION})
-
+if(WIN32)
+    install (FILES ${CMAKE_CURRENT_BINARY_DIR}/System.IO.Compression.Native-static.dir/$<CONFIG>/System.IO.Compression.Native-static.pdb DESTINATION ${STATIC_LIB_DESTINATION})
```
Comment (jkotas) [inline src/libraries/Native/Windows/System.IO.Compression.Native/CMakeLists.txt:126]: We need to binplace the .pdb as well for the lib to be actually useful. It is similar to what we do in other places.

File: eng/liveBuilds.targets [hunk 3]
```diff
@@ -149,6 +149,7 @@
       <LibrariesRuntimeFiles Include="
         $(LibrariesNativeArtifactsPath)*.dll;
         $(LibrariesNativeArtifactsPath)*.dylib;
+        $(LibrariesNativeArtifactsPath)*.lib;
```
Comment (jkotas) [inline eng/liveBuilds.targets:152]: Treat both .lib and .a same way for platform parity

File: src/coreclr/runtime.proj [hunk 4]
```diff
@@ -17,6 +17,7 @@
       <_CoreClrBuildArg Condition="$([MSBuild]::IsOsPlatform(Windows)) and
                                    ('$(TargetArchitecture)' == 'x86' or '$(TargetArchitecture)' == 'x64') and
                                    '$(Configuration)' == 'Release' and
+                                   '$(ClrRuntimeSubset)' == 'true' and
```
Comment (jkotas) [inline src/coreclr/runtime.proj:20]: cc @trylek This is fixing bogus pgocheck error that one gets when skipping build of runtime subset.  Introduced by #46019

File: src/libraries/Native/Windows/System.IO.Compression.Native/CMakeLists.txt (hunk 2 above)
Comment (VSadov) [reply src/libraries/Native/Windows/System.IO.Compression.Native/CMakeLists.txt:126]: Comment in `strip_symbols` mentions that `$<TARGET_PDB_FILE:tgt>` could be used, but it does not work with resource dlls. 
This is not a resource dll, so maybe  `TARGET_PDB_FILE` would work here?

Comment (VSadov) [reply src/libraries/Native/Windows/System.IO.Compression.Native/CMakeLists.txt]: hmm, it looks like `TARGET_PDB_FILE` does not work with static libs either. 
https://gitlab.kitware.com/cmake/cmake/-/issues/16932

Comment (jkotas) [reply src/libraries/Native/Windows/System.IO.Compression.Native/CMakeLists.txt]: I am going to omit this change from this PR.

**Review by VSadov — APPROVED** [review]
LGTM.

**Review by jkoritzinsky — APPROVED** [review]

**Review by MichalStrehovsky — APPROVED** [review]
Thank you!

**Review by trylek — APPROVED** [review]
Looks great, thanks for fixing!

**Pull request merged**